pkg/tsdb/cloudwatch/models/request_parser_test.go

```diff
@@ -124,255 +138,486 @@ func TestRequestParser(t *testing.T) {
+		query := []backend.DataQuery{
+			{
+				JSON: json.RawMessage(`{
+				   "refId":"ref1",
+				   "region":"us-east-1",
+				   "namespace":"ec2",
+				   "metricName":"CPUUtilization",
+				   "statistic":"Average",
+				   "period":"900",
+				   "type":"timeSeriesQuery"
+				}`),
+			},
+		}
+		res, err := ParseMetricDataQueries(query, time.Now().Add(-2*time.Hour), time.Now().Add(-time.Hour), false)
 		require.NoError(t, err)
-		assert.Equal(t, "ref1", res.RefId)
-		assert.Equal(t, "queryref1", res.Id)
+		require.Len(t, res, 1)
+		require.NotNil(t, res[0])
+		require.True(t, res[0].ReturnData)
 	})
-
-	t.Run("Valid id is generated if ID is not provided and refId is not a valid MetricData ID", func(t *testing.T) {
-		query := getBaseJsonQuery()
-		query.RefId = "$$"
-		res, err := parseRequestQuery(query, "$$", time.Now().Add(-2*time.Hour), time.Now().Add(-time.Hour))
+	t.Run("when hide is true, ReturnData is false", func(t *testing.T) {
+		query := []backend.DataQuery{
+			{
+				JSON: json.RawMessage(`{
+				   "refId":"ref1",
+				   "region":"us-east-1",
+				   "namespace":"ec2",
+				   "metricName":"CPUUtilization",
+				   "statistic":"Average",
+				   "period":"900",
+				   "type":"timeSeriesQuery",
+				   "hide":true
+				}`),
+			},
+		}
+		res, err := ParseMetricDataQueries(query, time.Now().Add(-2*time.Hour), time.Now().Add(-time.Hour), false)
 		require.NoError(t, err)
-		assert.Equal(t, "$$", res.RefId)
-		assert.Regexp(t, validMetricDataID, res.Id)
+		require.Len(t, res, 1)
+		require.NotNil(t, res[0])
+		require.False(t, res[0].ReturnData)
+	})
+	t.Run("when hide is false, ReturnData is true", func(t *testing.T) {
+		query := []backend.DataQuery{
+			{
+				JSON: json.RawMessage(`{
+				   "refId":"ref1",
+				   "region":"us-east-1",
+				   "namespace":"ec2",
+				   "metricName":"CPUUtilization",
+				   "statistic":"Average",
+				   "period":"900",
+				   "type":"timeSeriesQuery",
+				   "hide":false
+				}`),
+			},
+		}
+		res, err := ParseMetricDataQueries(query, time.Now().Add(-2*time.Hour), time.Now().Add(-time.Hour), false)
+		require.NoError(t, err)
+		require.Len(t, res, 1)
+		require.NotNil(t, res[0])
+		require.True(t, res[0].ReturnData)
+	})
+	t.Run("when query type is empty and hide is missing, ReturnData is true", func(t *testing.T) {
+		query := []backend.DataQuery{
+			{
+				JSON: json.RawMessage(`{
+				   "refId":"ref1",
+				   "region":"us-east-1",
+				   "namespace":"ec2",
+				   "metricName":"CPUUtilization",
+				   "statistic":"Average",
+				   "period":"900"
+				}`),
+			},
+		}
+		res, err := ParseMetricDataQueries(query, time.Now().Add(-2*time.Hour), time.Now().Add(-time.Hour), false)
+		require.NoError(t, err)
+		require.Len(t, res, 1)
+		require.NotNil(t, res[0])
+		require.True(t, res[0].ReturnData)
 	})
 
-	t.Run("parseRequestQuery sets label when label is present in json query", func(t *testing.T) {
```

Comment:
this test was shifted down to around line 597 https://github.com/grafana/grafana/pull/57431/files#diff-e2107889a319fc8692d34cb7ed8146ba186329754d1dcd33324ba4c29862713bR597